### ISSUE TITLE
LDAP errors when importing users

### ISSUE THREAD

**Issue by mwhamgenomics** [opencb/opencga#1676]
When importing LDAP users (i.e. `./opencga-admin.sh users import --auth-origin auth_origin -u user`), I get a combination of `java.net.UnknownHostException` and `javax.naming.InvalidNameException: [LDAP: error code 34 - invalid DN]`. I don't know why I sometimes get one and sometimes the other.

I stepped through the import in JDB, and it looks like LDAPAuthenticationManager is entering the `if` statement at line 243 and passing the string `"null"` as both the security principal and the credentials into the created dir context - see below for evidence of this.

This is happening even though I don't have values configured for `authUserId` or `authPassword` because at lines 73/74, it looks in the config, finds nothing and casts `null` into a String. These configs were added after #1604, so I assume these are optional in case an LDAP instance needs authentication to do searches?

If `authUserId` and `authPassword` are optional then we should fix the string casting and conditional statements. If they're required then these should be documented and added to the example configuration.yml.

JDB evidence:

Catching `java.net.UnknownHostException`:
```
> catch all java.net.UnknownHostException
...
Exception occurred: java.net.UnknownHostException (to be caught at: java.net.InetAddress.getAddressesFromNameService(), line=1,327 bci=62)"thread=main", java.net.Inet6AddressImpl.lookupAllHostAddr(), line=-1 bci=-1
main[1] up 28
main[29] print env
 env = "{java.naming.provider.url=ldaps://<ldap_server>:636, java.naming.factory.initial=com.sun.jndi.ldap.LdapCtxFactory, com.sun.jndi.ldap.connect.timeout=500, java.naming.security.principal=null, java.naming.security.authentication=simple, java.naming.security.credentials=null}"
```

Catching `javax.naming.InvalidNameException`:
```
> catch all javax.naming.InvalidNameException
...
Exception occurred: javax.naming.InvalidNameException (to be caught at: com.sun.jndi.ldap.LdapCtx.<init>(), line=320 bci=401)"thread=main", com.sun.jndi.ldap.LdapCtx.processReturnCode(), line=3,106 bci=864

main[1] locals
Method arguments:
res = instance of com.sun.jndi.ldap.LdapResult(id=3856)
resolvedName = null
resolvedObj = instance of com.sun.jndi.ldap.LdapCtx(id=3857)
remainName = null
envprops = instance of java.util.Hashtable(id=3858)
fullDN = null
Local variables:
msg = "[LDAP: error code 34 - invalid DN]"
r = null
e = instance of javax.naming.InvalidNameException(id=3855)
main[1] up 2 
main[3] locals
Method arguments:
startTLS = false
Local variables:
user = "null"
passwd = "null"
secProtocol = "ssl"
socketFactory = "javax.net.ssl.SSLSocketFactory"
authMechanism = "simple"
ver = null
usePool = false
initial = true
ldapVersion = 32
answer = instance of com.sun.jndi.ldap.LdapResult(id=3856)
```

**Comment by pfurio** [comment]
Thanks for the report.

**Comment by pfurio** [comment]
Hi @mwhamgenomics, I just pushed the fix. Could you try again and let me know if you find any other issues?

**Comment by mwhamgenomics** [comment]
Hi @pfurio, I confirm that this fixes the import. I'm still getting errors logging in though - will open a separate issue for that
